src/main/java/opendcs/util/sql/WrappedConnection.java

```diff
@@ -0,0 +1,394 @@
+package opendcs.util.sql;
+
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.CallableStatement;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.SQLClientInfoException;
+import java.sql.SQLException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Savepoint;
+import java.sql.Statement;
+import java.sql.Struct;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+
+import javax.management.openmbean.CompositeData;
+import javax.management.openmbean.CompositeDataSupport;
+import javax.management.openmbean.OpenDataException;
+
+import decodes.cwms.CwmsConnectionPool;
+import ilex.util.Logger;
+import opendcs.org.opendcs.jmx.ConnectionPoolMXBean;
+import opendcs.org.opendcs.jmx.connections.JMXTypes;
+import opendcs.util.functional.ThrowingConsumer;
+
+/**
+ * Used for pooled connection system. Allows code
+ * to otherwise write a normal try-with-resources
+ * to close the connection but will correctly call
+ * theDb.freeConnection on behalf of the user
+ * 
+ * Otherwise it's just a passthrough to the realConnection
+ */
+public class WrappedConnection implements Connection{
+    private static Logger log = Logger.instance();
+
+    private Connection realConnection;
+    private final ThrowingConsumer<Connection,SQLException> onClose;
+    private Optional<CwmsConnectionPool> pool;
+    private List<StackTraceElement> openTrace = null;
+    private List<StackTraceElement> closeTrace = null;
+    private ZonedDateTime start = ZonedDateTime.now();
+    private ZonedDateTime end = null;
+
+    
+
+    public WrappedConnection(Connection realConnection){
+        this(realConnection, (c) -> {},Optional.ofNullable(null));
+    }
+
+    public WrappedConnection(Connection realConnection, final ThrowingConsumer<Connection,SQLException> onClose, Optional<CwmsConnectionPool> pool )
```

Comment:
`Optional` pool is strange to me. How would the connection not be null, but the pool it came from not be null. Also this is an anti-pattern for the usage of Optional, it shouldn't be used as a parameter.

Reply:
Yeah that idea makes sense, but weird dependency-wise having the `opendcs` package depend on a `cwms` package. `private Optional<ConnectionPool> getConnectionPool() { Optional.ofNullable(this.pool);}` would work for this case.

Reply:
Good point, actually I think it can just be dropped in favor of a simple boolean. The original intent had it actually using the object but now it's just don't do extra operations unless told.

Reply:
cleaned up.